testament/categories.nim

```diff
@@ -448,7 +448,10 @@ proc testNimblePackages(r: var TResults; cat: Category; packageFilter: string) =
           (outp, status) = execCmdEx(cmd, workingDir = workingDir2)
           status == QuitSuccess
         if not ok:
-          addResult(r, test, targetC, "", cmd & "\n" & outp, reFailed)
+          if pkg.allowFailure:
+            inc r.passed
```

Comment:
should it really count as passed though? this way the counts are inaccurate

Reply:
I know, see this note in the PR content:
```
## xxx rename passed to passedOrAllowedFailure
```

note that the counts can be deduced since we have both passed (which we can rename to passedOrAllowedFailure) and failedButAllowed, from which you can deduce each count.

future work can report the counts in a clearer way.